db/migrate/20200203200855_add_ancestry_to_miq_ae_namespace.rb

```diff
@@ -0,0 +1,40 @@
+class AddAncestryToMiqAeNamespace < ActiveRecord::Migration[5.0]
+  class MiqAeNamespace < ActiveRecord::Base
+    self.inheritance_column = :_type_disabled # disable STI
+  end
+
+  def update_ns_parent(parent)
+    ancestry = [parent.ancestry, parent.id].compact.join("/") if parent
+    MiqAeNamespace.where(:parent_id => parent.try(:id)).each do |ns|
+      ns.update(:ancestry => ancestry) if ancestry
+      update_ns_parent(ns)
+    end
+  end
+
+  def up
+    add_column :miq_ae_namespaces, :ancestry, :string
+    add_index :miq_ae_namespaces, :ancestry
+
+    say_with_time("Converting MiqAeNamespaces from parent_id to ancestry") do
+      update_ns_parent(nil)
+    end
+
+    remove_index :miq_ae_namespaces, :column => :parent_id
+    remove_column :miq_ae_namespaces, :parent_id
+  end
+
+  def down
+    add_column :miq_ae_namespaces, :parent_id, :bigint
+    add_index :miq_ae_namespaces, :parent_id
+
+    say_with_time("Converting MiqAeNamespaces from ancestry to parent_id") do
+      MiqAeNamespace.all.each do |ns|
+        parent_ns_id = ns.ancestry.split("/").last.to_i if ns.ancestry.present?
+        ns.update(:parent_id => parent_ns_id)
+      end
```

Comment:
I think this can be done in a single query... looking for my original code change.